hyperspectral/hyperspectral_indices_make.nco

```diff
@@ -1,294 +1,555 @@
+  
+   flg_mss - set the _FillValue 1.0e36 to all values in selected reflectances where value <=0.0f (default 1)
+
+   usage examples 
+   ncap2 -v -O  -S hyperspectral_indices_make.nco   in.nc out.nc                                         ( use default flags)     
+   ncap2 -v -O -s 'flg_dbg=1s;flg_rba=0s;flg_mss=1s;' -S hyperspectral_indices_make.nco   in.nc out.nc   ( set flags )
+
+
+*/
+   
 
-iR600=min_coords(wavelength, 6e-7d);
-iR650=min_coords(wavelength, 6.5e-7d);
-iR670=min_coords(wavelength, 6.7e-7d);
-iR680=min_coords(wavelength, 6.8e-7d);
-iR690=min_coords(wavelength, 6.9e-7d);
 
-iR700=min_coords(wavelength, 7.00e-7d);
-iR705=min_coords(wavelength, 7.05e-7d);
-iR710=min_coords(wavelength, 7.1e-7d);
-iR720=min_coords(wavelength, 7.2e-7d);
-iR740=min_coords(wavelength, 7.4e-7d);
-iR750=min_coords(wavelength, 7.5e-7d);
-iR760=min_coords(wavelength, 7.6e-7d);
-iR780=min_coords(wavelength, 7.8e-7d);
-iR790=min_coords(wavelength, 7.9e-7d);
+if(!exists(flg_dbg))
+   *flg_dbg=0s;  // [flg] Print debug info
 
+if(!exists(flg_rba))
+   *flg_rba=1s;  // [flg] calculate ratios before average
 
-iR800=min_coords(wavelength, 8e-7d);
+if(!exists(flg_mss)){
+  *flg_mss=1s;  // [flg] add and set _FillValue to reflectance indices
+  flt_mss=1.0e36f;
+}
+
+
+
+
+//required reflectances
+@rfl_lst={"R445"s,"R450"s,"R470"s, "R500"s,"R512"s,"R531"s, "R540"s,"R550"s,"R570"s,"R586"s,"R590"s,"R600"s,"R650"s,"R670"s, "R680"s,"R690"s,"R700"s,"R705"s,"R710"s,"R720"s,"R740"s,"R750"s,"R760"s, "R780"s,"R790"s,"R800"s,"R900"s,"R970"s};  
+
+// nb ONE-TO_ONE  correspondence with above list
+@dbl_val_lst={ 4.45e-7d, 4.5e-7d, 4.7e-7d, 5.0e-7d, 5.12e-7, 5.31e-7, 5.4e-7, 5.5e-7, 5.7e-7,5.86e-7, 5.9e-7,6.0e-7,6.5e-7,6.7e-7, 6.8e-7, 6.9e-7, 7.0e-7, 7.05e-7, 7.1e-7,7.2e-7, 7.4e-7,7.5e-7,7.6e-7, 7.8e-7, 7.9e-7, 8.0e-7, 9.0e-7,9.7e-7};    
+
+// use vpointers to calculate averages AFTER pixel ratios
+@ind_lst={"NDVI"s,"SR"s, "OSAVI"s,"CHL"s,"msR705"s,"TCARI"s,"CarChap"s,"Car1Black"s,"Car2Black"s,"PRI570"s,"SIPI"s,"antGamon"s,"antGitelson"s,"CHLDela"s,"CI"s,"PRI586"s,"PRI512"s,"FRI1"s,"FRI2"s,"NDVI1"s,"RERI"s,"ZM"s,"REP"s,"NDRE"s,"TVI"s};
+
+
+
+/* sanity check */
+if( @rfl_lst.size() != @dbl_val_lst.size() )
+{
+  print("size difference between list @rfl_lst and @dbl_val_lst\n");
+  exit(1);
+}
+
+
+/* extract the required reflectances - 
+   nb min_coords() -extracts the index of the grid value nearest the required value 
+   This step creates the vars iR445, iR450, iR470..iR970 */
+{
+  *idx=0;
+  *sz=@rfl_lst.size();
+
+  for(idx=0;idx<sz;idx++)
+  {
+    @R_nm=sprint(@rfl_lst(idx));
+    @iR_nm=push("i",@R_nm);
+
+    *@iR_nm=min_coords(wavelength,@dbl_val_lst(idx));
+          
+  }
+
+
+}
 
-iR900=min_coords(wavelength, 9e-7d);
-iR970=min_coords(wavelength, 9.7e-7d);
 
 
-// sanity check for wavelength coord  check min ,max, mode 
+/* sanity checks for required relectances -  make sure the are within resonable tolerance (1.0e-9) */
```

Comment:
HI David,
for i each index I can check if the required reflectances are available 
Would this be OK ?